scripts/tests/run_tv_casting_test.py

```diff
@@ -0,0 +1,305 @@
+    logging.error('Discovery failed!')
+
+    for log_file_path in log_file_paths:
+        dump_logs_to_console(log_file_path)
+
+    sys.exit(1)
+
+
+# Extract and return an integer value from a given output string.
+#   line: The string containing the integer value.
+#
+# The string is expected to be in the following format as it is received
+# from the Linux tv-casting-app output:
+#   \x1b[0;34m[1713741926895] [7276:9521344] [DIS] Vendor ID: 65521\x1b[0m
+# The integer value to be extracted here is 65521.
+def extract_value_from_string(line: str) -> int:
+    value = line.split(':')[-1].strip().replace('\x1b[0m', '')
+    value = int(value)
+
+    return value
+
+
+# Validate if the discovered value matches the expected value.
+#   expected_value: The expected integer value, i.e. any of the VENDOR_ID,
+#                   PRODUCT_ID, or DEVICE_TYPE constants.
+#   line: The string containing the value of interest that will be compared
+#         to the expected value.
+#   value_name: The name of the discovered value, i.e. 'Vendor ID', 'Product ID',
+#               or 'Device Type'.
+#   Return False if the discovered value does not match, True otherwise.
+def validate_value(expected_value: int, line: str, value_name: str) -> bool:
+    # Extract the integer value from the string.
+    value = extract_value_from_string(line)
+
+    # If the discovered value does not match the expected value,
+    # log the error and return False.
+    if value != expected_value:
+        logging.error(f'{value_name} does not match the expected value!')
+        logging.error(f'Expected {value_name}: {expected_value}')
+        logging.error(line.rstrip('\n'))
+        return False
+
+    # Return True if the value matches the expected value.
+    return True
+
+
+# Test if the Linux tv-casting-app is able to discover the Linux tv-app. Both will
+# run separately as subprocesses, with their outputs written to respective log files.
+# Default paths for the executables are provided but can be overridden via command line
+# arguments. For example: python3 run_tv_casting_test.py --tv-app-rel-path=path/to/tv-app
+#                         --tv-casting-app-rel-path=path/to/tv-casting-app
+@click.command()
+@click.option('--tv-app-rel-path', type=str, default='out/tv-app/chip-tv-app', help='Path to the Linux tv-app executable.')
+@click.option('--tv-casting-app-rel-path', type=str, default='out/tv-casting-app/chip-tv-casting-app', help='Path to the Linux tv-casting-app executable.')
+def test_discovery_fn(tv_app_rel_path, tv_casting_app_rel_path):
+
+    # Store the log files to a temporary directory.
+    with tempfile.TemporaryDirectory() as temp_dir:
+        linux_tv_app_log_path = os.path.join(temp_dir, LINUX_TV_APP_LOGS)
+        linux_tv_casting_app_log_path = os.path.join(temp_dir, LINUX_TV_CASTING_APP_LOGS)
+
+        # Open and write to the log file for the Linux tv-app.
+        with LogFileManager(linux_tv_app_log_path, 'w') as linux_tv_app_log_file:
```

Comment:
You could create a compund with for both log paths, so that things do not indent as much:

https://docs.python.org/3/reference/compound_stmts.html#with

https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory seems to indicate that `delete=True` is the default and cleanup will be done automatically. If so why do we need a separate class that cleans thigns up? Could we just do 

```py
with open(linux_tv_app_log_path, 'w') as linux_tv_app_log_file, open(linux_tv_casting_app_log_path, 'w') as linux_tv_casting_app_log_file:
```

Reply:
Good catch! I have removed the LogFileManager class as the temporary directory will take care of the cleanup.